src/Components/TodoHeader.tsx

```diff
@@ -0,0 +1,133 @@
+import { FormEvent, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { Notification } from '../types/Notification';
+import { createTodo, getTodos, updateTodo } from '../api/todos';
+import { USER_ID } from '../utils/constants';
+
+interface Props {
+  todos: Todo[],
+  setTodos: (todo: Todo[]) => void,
+  setErrorMessage: (value: string) => void,
+  setTempTodo: (value: Todo | null) => void,
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>,
+}
+
+export const TodoHeader: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setErrorMessage,
+  setTempTodo,
+  setLoadingIds,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const completedTodosCount = todos.filter(todo => todo.completed).length;
+
+  const handleFormSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (title.trim() === '') {
+      setErrorMessage(Notification.title);
+
+      return;
+    }
+
+    const newTodo = { title, userId: USER_ID, completed: false };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    createTodo(newTodo)
+      .then(() => {
+        getTodos(USER_ID)
+          .then((value) => {
+            setTodos(value);
+            setTempTodo(null);
+          });
+      })
+      .catch(() => {
+        setErrorMessage(Notification.add);
+      });
+
+    setTitle('');
+  };
+
+  const handleToggleAll = () => {
+    const uncompletedTodos = todos.filter(todo => todo.completed === false);
+    const uncompletedTodosIds = uncompletedTodos.map(todo => todo.id);
+
+    if (uncompletedTodos.length === 0) {
+      setLoadingIds(currIds => [...currIds, ...todos.map(todo => todo.id)]);
+
+      Promise.all(todos.map(todo => {
+        return updateTodo(todo.id, { completed: !todo.completed });
+      }))
+        .then(() => {
+          getTodos(USER_ID)
+            .then((value) => {
+              setTodos(value);
+              setLoadingIds([]);
+            })
+            .catch(() => {
+              setErrorMessage(Notification.load);
+            });
+        })
+        .catch(() => {
+          setErrorMessage(Notification.update);
+        });
+
+      return;
+    }
+
+    setLoadingIds(currIds => [...currIds, ...uncompletedTodosIds]);
+
+    Promise.all(uncompletedTodos.map(todo => {
+      return updateTodo(todo.id, { completed: !todo.completed });
+    }))
+      .then(() => {
+        getTodos(USER_ID)
+          .then((value) => {
+            setTodos(value);
+          })
```

Comment:
same. suggest do not make request to server and just update locally todos.
check other similar cases